projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/datajobs/DataJobsExecutionControllerIT.java

```diff
@@ -44,6 +53,20 @@ public void testDataJobExecutionStartNotFound() throws Exception {
   }
 
   // TODO: test all methods
+  @Test
+  @WithMockUser
+  @DirtiesContext
+  public void testWhenADeploymentFailsNoEntryIsSavedInTheDatabase() throws Exception {
+    // arrange
+    doThrow(new RuntimeException())
+        .when(dataJobsKubernetesService)
+        .startNewCronJobExecution(any(), any(), any(), any(), any(), any());
+    // act
+    TestUtils.createDataJob(mockMvc, TEST_TEAM_NAME, TEST_JOB_NAME);
+    TestUtils.createDeployment(mockMvc, TEST_TEAM_NAME, TEST_JOB_NAME);
```

Comment:
I am not sure the idea of the test but without starting execution, the changed code would never be called. 

Maybe there's some misunderstanding in terminology: 

See https://github.com/vmware/versatile-data-kit/wiki/User-Guide#data-jobs-development-workflow or https://raw.githubusercontent.com/wiki/vmware/versatile-data-kit/vdk-data-job-lifecycle-state-diagram.png 

Basically the user workflow is like this 
1. User creates a job using [Control Service Job API ](https://app.swaggerhub.com/apis-docs/aivanovorg/taurus-data-pipelines/1.1#/Data%20Jobs)
   Then they develop it and test it locally 
2. Eventually they would deploy it to run in our managed by Control Service runtime using CS [Deployment API](https://app.swaggerhub.com/apis-docs/aivanovorg/taurus-data-pipelines/1.1#/Data%20Jobs%20Deployment) .  Deployment will build, test, version/release and deploy the job in Kubernetes. The job is ready to be executed but it' not yet.
3. If schedule is specified the job would be executed every so often and Exectuion Servie will track it. Users can request recent executions or start a manual execution using [Execution API](https://app.swaggerhub.com/apis-docs/aivanovorg/taurus-data-pipelines/1.1#/Data%20Jobs%20Execution) 

So in reality our Control Service is a collection of a few (fairly) independent services (Job, Deployment and Execution among them). 

In this mock tests, the mock deployment only deploys a job without any schedule. So to test anything done by Execution Service we need to explicitly call some method like TestUtils.startMockExecution .


Of course it's possible that I am the one missing something in this particular case. But still want to find out what.

Reply:
Nope your not missing anything. 
This was a mistake on my side. 
Fixed now.